### PR TITLE
Fibonacci Number [Java]

### LINKED ISSUE #444: Fibonacci Number [Java]
#### This issue is about [Fibonacci Number](http://www.geeksforgeeks.org/program-for-nth-fibonacci-number/).

- [X] Issue Title is in form `Algo/DS Name [Language]`.
- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [X] This Algo/DS is worth adding to this repository.
- [X] Assign this issue to me (I want to work on this).
	- [X] I ensure that I am not already assigned to 2 or more issues.
- [X] I will send my PR only after I'm assigned to this issue by any maintainer.
- [X] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by shikherjaiswal**
Fixes #444 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once the Travis build passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

### CODE REVIEW

**Comment by shikherjaiswal** [comment]
Please review @Monal5031  @mohitkyadav

**Comment by shikherjaiswal** [comment]
I think travis build is not related to my changes . Please review .

File: fibonacci_number/FibonacciNumber.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        for(int i = 2; i < n; i++) {
+          fib = a + b;
+          a = b;
+          b = fib;
+        }
+        System.out.println(fib);
+      }
+    }
+
+      public static void main(String[] args) {
```
Comment (mohitkyadav) [inline fibonacci_number/FibonacciNumber.java]: @shikherjaiswal indentation is a  little off in this block.
5 space expected 4.

**Comment by shikherjaiswal** [comment]
@mohitkyadav  i have made the changes. Please review

**Comment by mohitkyadav** [comment]
@shikherjaiswal update your branch.

**Comment by shikherjaiswal** [comment]
I have updated the branch and changed the base case to suggested alternative

**Comment by shikherjaiswal** [comment]
I have updated the branch . Please review

**Comment by aashutoshrathi** [comment]
@shikherjaiswal Please indent your code properly.
Expected indentation for Java is of size 4.

File: fibonacci_number/FibonacciNumber.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+public class FibonacciNumber {
+
+    public static void fibonacci(int n) {
```
Comment (singhpratyush) [inline fibonacci_number/FibonacciNumber.java]: Please make the function to return the result and print it in the `main` method only, just to keep it consistent with the rest of the code.

File: fibonacci_number/FibonacciNumber.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+public class FibonacciNumber {
+
+    public static int fibonacci(int n) {
+        if (n == 1 || n == 2) {
+            return (n - 1);
+        } else {
+            int a = 0; // Store second last fibonacci number
```
Comment (mohitkyadav) [inline fibonacci_number/FibonacciNumber.java]: @shikherjaiswal Please do the comments above the related code.
```
// a comment
doSomething();
```

File: fibonacci_number/FibonacciNumber.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+public class FibonacciNumber {
+
+    public static int fibonacci(int n) {
```
Comment (mohitkyadav) [inline fibonacci_number/FibonacciNumber.java]: also do a comment related to the function.
```
// this function does this
function (args[]) {

}
```

**Pull request merged**